metadata-ingestion/src/datahub/utilities/file_backed_collections.py

```diff
@@ -1,87 +1,162 @@
 import collections
+import pathlib
 import sqlite3
-import tempfile
 from dataclasses import dataclass, field
 from typing import (
     Any,
     Callable,
+    Dict,
     Generic,
     Iterator,
     List,
     MutableMapping,
+    Optional,
     OrderedDict,
     Sequence,
     Tuple,
     TypeVar,
     Union,
 )
 
+_DEFAULT_TABLE_NAME = "data"
+_DEFAULT_MEMORY_CACHE_MAX_SIZE = 2000
```

Comment:
do we know roughly how much ram this will consume?

Reply:
it depends on the actual size of the objects stored within

assuming avg 20kb per object, 2000 * 20kb -> 40mb, which is totally reasonable